### PR TITLE
[erdjs] add tx options field

### PR DESCRIPTION
**Pull request by bogdan-rosianu**
added the field `options` for transactions. 
In the protocol, it is treated as `omitempty` so if it's value in `0` then it won't be included in the payload that is to be signed (added a unit test that ensures this)

### CODE REVIEW

**Review by camilbancioiu — DISMISSED** [review]

File: erdjs/src/transaction.spec.ts [hunk 1]
```diff
@@ -1,11 +1,11 @@
-import { assert } from "chai";
-import { Transaction } from "./transaction";
+import {assert} from "chai";
```
Comment (andreibancioiu) [inline erdjs/src/transaction.spec.ts:1]: The whitespace changed from `{ foo }` to `{foo}`. Is this intended?

File: erdjs/src/networkParams.ts [hunk 2]
```diff
@@ -138,6 +138,30 @@ export class TransactionVersion {
     }
 }
 
+export class TransactionOptions {
```
Comment (andreibancioiu) [inline erdjs/src/networkParams.ts:141]: Not sure - perhaps on this class we should add construction utilities such as `TransactionOptions.fromFlags({ signHashInsteadOfContent: true, fooBar: false })` - not necessarily this way, and not necessarily now - but currently, the construction from a raw number (one might ask _what is about this number?_) isn't very meaningful.

Does this make sense?

**Review by andreibancioiu — DISMISSED** [review]

File: erdjs/src/transaction.spec.ts (hunk 1 above)
Comment (bogdan-rosianu) [reply erdjs/src/transaction.spec.ts:1]: yes, the linter changed that for me after re-formatting the code

File: erdjs/src/networkParams.ts (hunk 2 above)
Comment (bogdan-rosianu) [reply erdjs/src/networkParams.ts:141]: Yes, will add also a private constructor (`.fromFlags`)

Comment (bogdan-rosianu) [reply erdjs/src/networkParams.ts:160]: Done. created 'static' constructors such as `withDefaultVersion` or `withTxHashSigningOptions` for both TransactionVersion and TransactionOptions

File: erdjs/src/transaction.ts [hunk 3]
```diff
@@ -15,8 +15,8 @@ import {ProtoSerializer} from "./proto";
 
 const createTransactionHasher = require("blake2b");
 
-const TRANSACTION_VERSION = new TransactionVersion(1);
-const TRANSACTION_OPTIONS = new TransactionOptions(0);
+const TRANSACTION_VERSION = TransactionVersion.withDefaultVersion();
+const TRANSACTION_OPTIONS = TransactionOptions.withDefaultOptions();
```
Comment (camilbancioiu) [inline erdjs/src/transaction.ts:19]: Suggest renaming the consts to `DEFAULT_TRANSACTION_VERSION` and `DEFAULT_TRANSACTION_OPTIONS`, if it makes sense for their use.

Comment (bogdan-rosianu) [reply erdjs/src/transaction.ts]: done

**Review by camilbancioiu — DISMISSED** [review]

File: erdjs/src/networkParams.spec.ts [hunk 4]
```diff
@@ -0,0 +1,40 @@
+import {assert} from "chai";
```
Comment (andreibancioiu) [inline erdjs/src/networkParams.spec.ts:1]: :+1:

File: erdjs/src/networkParams.ts [hunk 5]
```diff
@@ -125,14 +130,66 @@ export class TransactionVersion {
...
+    /**
+     * Creates a TransactionOptions object with the default options setting
+     */
+    static withDefaultOptions(): TransactionOptions {
+        return new TransactionOptions(TRANSACTION_OPTIONS_DEFAULT);
+    }
+
+    /**
+     * Created a TransactionsOptions object with the setting for hash signing
+     */
```
Comment (andreibancioiu) [inline erdjs/src/networkParams.ts:188]: :+1:

**Review by andreibancioiu — DISMISSED** [review]

**Review by camilbancioiu — APPROVED** [review]

**Review by andreibancioiu — APPROVED** [review]

**Pull request merged**